vpr/src/analysis/timing_reports.cpp

```diff
@@ -10,6 +17,121 @@
 
 #include "VprTimingGraphResolver.h"
 
+/**
+ * @brief Get the bounding box of a net.
+ * If the net is completely absorbed into a cluster block, return the bounding box of the cluster block.
+ * Otherwise, return the bounding box of the net's route tree.
+ * If a net is not routed, bounding box is returned with default values (OPEN).
+ * 
+ * @param atom_net_id The id of the atom net to get the bounding box of.
+ */
+static t_bb get_net_bounding_box(const AtomNetId atom_net_id) {
+    const auto& route_trees = g_vpr_ctx.routing().route_trees;
+    const auto& rr_graph = g_vpr_ctx.device().rr_graph;
+
+    // Lambda to get the bounding box of a route tree
+    auto route_tree_bb = [&](const RouteTree& route_tree) {
+        t_bb bb;
+
+        // Set the initial bounding box to the root node's location
+        RRNodeId route_tree_root = route_tree.root().inode;
+        bb.xmin = rr_graph.node_xlow(route_tree_root);
+        bb.xmax = rr_graph.node_xhigh(route_tree_root);
+        bb.ymin = rr_graph.node_ylow(route_tree_root);
+        bb.ymax = rr_graph.node_yhigh(route_tree_root);
+        bb.layer_min = rr_graph.node_layer(route_tree_root);
+        bb.layer_max = rr_graph.node_layer(route_tree_root);
+
+        // Iterate over all nodes in the route tree and update the bounding box
+        for (auto& rt_node : route_tree.all_nodes()) {
+            RRNodeId inode = rt_node.inode;
+
+            if (rr_graph.node_xlow(inode) < bb.xmin)
+                bb.xmin = rr_graph.node_xlow(inode);
+            if (rr_graph.node_xhigh(inode) > bb.xmax)
+                bb.xmax = rr_graph.node_xhigh(inode);
+
+            if (rr_graph.node_ylow(inode) < bb.ymin)
+                bb.ymin = rr_graph.node_ylow(inode);
+            if (rr_graph.node_yhigh(inode) > bb.ymax)
+                bb.ymax = rr_graph.node_yhigh(inode);
+
+            if (rr_graph.node_layer(inode) < bb.layer_min)
+                bb.layer_min = rr_graph.node_layer(inode);
+            if (rr_graph.node_layer(inode) > bb.layer_max)
+                bb.layer_max = rr_graph.node_layer(inode);
```

Comment:
This lines can be more succinctly written as bb.xmin = min(bb.xmin, rr_graph.node_slow(inode)) etc. (will cut lines in half and be a bit easier to read).